src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,112 @@
+module Algebra.Graph.Acyclic.AdjacencyMap where
+
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AMA
+import qualified Algebra.Graph.AdjacencyMap.Internal as AMI
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.State.Strict
+import Data.Map (Map, (!))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show)
+
+consistent :: Ord a => AdjacencyMap a -> Bool
+consistent (AAM m) = AMI.consistent m && AMA.isAcyclic m
+
+type AdjacencyMapP a = AdjacencyMap (Int, a)
+
+empty :: AdjacencyMap a
+empty = AAM AM.empty
+
+vertex :: a -> AdjacencyMap a
+vertex x = AAM $ AM.vertex x
+
+vertices :: (Ord a) => [a] -> AdjacencyMap a
+vertices xs = AAM $ AM.vertices xs
+
+overlayD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+overlayD (AAM a) (AAM b) = AAM (AM.overlay (AM.gmap Left a) (AM.gmap Right b))
+
+connectD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+connectD (AAM a) (AAM b) = AAM (AM.connect (AM.gmap Left a) (AM.gmap Right b))
+
+transitiveClosure :: Ord a => AdjacencyMap a -> AdjacencyMap a
+transitiveClosure (AAM x) = AAM (AM.transitiveClosure x)
+
+overlay :: (Ord a) => AdjacencyMap a -> AdjacencyMap a -> AdjacencyMap a
```

Comment:
This can produce cyclic graphs, so shouldn't be here. There should be no functions that allow the user to create `Acyclic.AdjacencyMap` that is not `consistent`.

Reply:
Oh! I'm sorry, will change that ASAP!

Reply:
Here is an example: `1 * 2 + 2 * 1`. Both `1*2` and `2*1` are `consistent`, but the result isn't.